qiskit/compiler/transpiler.py

```diff
@@ -660,6 +660,9 @@ def _parse_transpile_args(
     }
 
     list_transpile_args = []
+    if scheduling_method is None and hasattr(backend, "get_scheduling_stage"):
+        scheduling_method = backend.get_scheduling_stage()
```

Comment:
Is there a use for a value of `scheduling_method` that means "don't schedule, no matter what"?  Previously it was possible to do that (give a backend that wants scheduling, but don't supply `scheduling_method`), and now it's not.

I don't know one way or the other, just pointing out the loss.  Perhaps more generally, a `transpile` argument `use_backend_passes` that defaults to `True`?

Reply:
That's a fair point, we probably should give users an option to disable this functionality if they don't want the backend to change the default behavior. I'll add this flag to `transpile()` and `generate_preset_pass_manager()` to provide the lever and also add an upgrade release note about the potential change in behavior.

Reply:
Simply we can set `None` if we don't want scheduling and I think this is default behavior (anyways the transpiler checks violation of time constraints, i.e. duration of delay instruction).

Reply:
Well that's kind of what we're getting at here, this potentially changes the behavior of the scheduling phase if `None` the default is used for `scheduling_method`. If the backend supplies it's own scheduling stage pass manager that gets used for `None` by default now, which could potentially include scheduling. The use case I had in mind was a backend running DD by default if it had sufficient knowledge on how best to run it, but with the way this is implemented right now there would be no way for a user to disable scheduling (or any of the backend provided stages) if they didn't want those running

Reply:
I added a new kwarg to ignore backend default methods in: https://github.com/Qiskit/qiskit-terra/pull/8648/commits/cf6a7ec90ab8b8b150abb92cce15a7935a0b6994 this way if a user doesn't want the backend to enable scheduling by default (despite the backend object doing that) this will let the user skip that (of course as before manually specifying an alternative method will still supersede the backend default).